### PR TITLE
PAYARA-2848 catch ConcurrentModificationException

### LINKED ISSUE #2833: ConcurrentModificationException in BeanDeploymentArchiveImpl.parseBeansXML
# Description #
----------

The cleanup step in `BeanDeploymentArchiveImpl.parseBeansXML` may fail when when `JarFileFactory` is modifiying its cache at that time. And that tend to happen during deployment.

## Expected Outcome

Deployment should not fail due to factors not related to the deployment, `ConcurrentModificationException` should be caught and ignored.

## Current Outcome

````
11:28:29.008 [main] ERROR javax.enterprise.system.tools.deployment.common - Exception while invoking class org.glassfish.weld.WeldDeployer load method java.util.ConcurrentModificationException: null
	at java.util.HashMap$HashIterator.nextNode(HashMap.java:1437)
	at java.util.HashMap$ValueIterator.next(HashMap.java:1466)
	at java.util.AbstractCollection.addAll(AbstractCollection.java:343)
	at org.glassfish.weld.BeanDeploymentArchiveImpl.parseBeansXML(BeanDeploymentArchiveImpl.java:782)
	at org.glassfish.weld.BeanDeploymentArchiveImpl.populate(BeanDeploymentArchiveImpl.java:541)
	at org.glassfish.weld.BeanDeploymentArchiveImpl.<init>(BeanDeploymentArchiveImpl.java:161)
	at org.glassfish.weld.BeanDeploymentArchiveImpl.populate(BeanDeploymentArchiveImpl.java:526)
	at org.glassfish.weld.BeanDeploymentArchiveImpl.<init>(BeanDeploymentArchiveImpl.java:161)
	at org.glassfish.weld.RootBeanDeploymentArchive.createModuleBda(RootBeanDeploymentArchive.java:97)
	at org.glassfish.weld.RootBeanDeploymentArchive.<init>(RootBeanDeploymentArchive.java:90)
	at org.glassfish.weld.DeploymentImpl.createModuleBda(DeploymentImpl.java:731)
	at org.glassfish.weld.DeploymentImpl.<init>(DeploymentImpl.java:145)
	at org.glassfish.weld.WeldDeployer.load(WeldDeployer.java:528)
	at org.glassfish.weld.WeldDeployer.load(WeldDeployer.java:112)
	at org.glassfish.internal.data.ModuleInfo.load(ModuleInfo.java:209)
	at org.glassfish.internal.data.ApplicationInfo.load(ApplicationInfo.java:318)
	at com.sun.enterprise.v3.server.ApplicationLifecycle.deploy(ApplicationLifecycle.java:497)
	at com.sun.enterprise.v3.server.ApplicationLifecycle.deploy(ApplicationLifecycle.java:220)
	at org.glassfish.deployment.admin.DeployCommand.execute(DeployCommand.java:508)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$2$1.run(CommandRunnerImpl.java:544)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$2$1.run(CommandRunnerImpl.java:540)
	at java.security.AccessController.doPrivileged(Native Method)
	at javax.security.auth.Subject.doAs(Subject.java:360)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$2.execute(CommandRunnerImpl.java:539)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$3.run(CommandRunnerImpl.java:570)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$3.run(CommandRunnerImpl.java:562)
	at java.security.AccessController.doPrivileged(Native Method)
	at javax.security.auth.Subject.doAs(Subject.java:360)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl.doCommand(CommandRunnerImpl.java:561)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl.doCommand(CommandRunnerImpl.java:1469)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl.access$1300(CommandRunnerImpl.java:111)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$ExecutionContext.execute(CommandRunnerImpl.java:1851)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$ExecutionContext.execute(CommandRunnerImpl.java:1727)
	at com.sun.enterprise.admin.cli.embeddable.DeployerImpl.deploy(DeployerImpl.java:134)
````

## Steps to reproduce (Only for bug reports) 

1. Deploy complex web app
2. Be lucky

We manage to see it time to time in our tests with Embedded Payara but the nature of the bug does not allow for creating a reproducer.

## Environment ##

- **Payara Version**: 4.1.2.181
- **Edition**: Full (Embedded)
- **JDK Version**:  8u144
- **Operating System**: Windows
- **Database**:  N/A

### PR DESCRIPTION
**Pull request by svendiedrichsen**
Fixes #2833

### CODE REVIEW

**Comment by smillidge** [comment]
Jenkins test please

**Comment by svendiedrichsen** [comment]
@smillidge I'm trying to put this functionality somewhere but I don't know where. This code to close JarFiles cached exists 3 times in the codebase and I would like to fix it for all of them.

**Comment by svendiedrichsen** [comment]
And also there is a chance of avoiding the ConcurrentModificationException by synchronizing against the cached JarFileFactory instance. This allows to fully iterate and close all JarFiles.

**Comment by svendiedrichsen** [comment]
Jenkins test please

File: nucleus/admin/util/src/main/java/com/sun/enterprise/admin/util/JarFileUtils.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
+package com.sun.enterprise.admin.util;
```
Comment (smillidge) [inline nucleus/admin/util/src/main/java/com/sun/enterprise/admin/util/JarFileUtils.java]: This file needs the Payara copyright header

Comment (svendiedrichsen) [reply nucleus/admin/util/src/main/java/com/sun/enterprise/admin/util/JarFileUtils.java]: Sure. I'll fix this.

**Comment by svendiedrichsen** [comment]
@smillidge Done.

**Comment by smillidge** [comment]
thanks

**Comment by smillidge** [comment]
jenkins test please

**Pull request merged**